crates/fuzzing/wasm-spec-interpreter/build.rs

```diff
@@ -0,0 +1,42 @@
+/// Build the OCaml code and statically link it into the Rust library; see the [ocaml-interop
+/// example](https://github.com/tezedge/ocaml-interop/blob/master/testing/rust-caller/build.rs) for
+/// more details. After playing with this a bit, I discovered that the best approach to avoid
+/// missing symbols was to imitate `dune`: I observed `rm -rf _build && dune build
+/// ./ocaml/interpret.exe.o --display=verbose` and used that as a pattern, now encoded in
+/// `ocaml/Makefile` for easier debugging.
+use std::{env, process::Command};
+
+const LIB_NAME: &'static str = "interpret";
+const OCAML_DIR: &'static str = "ocaml";
+
+fn main() {
+    let out_dir = &env::var("OUT_DIR").unwrap();
+
+    // Re-run if changed.
+    println!("cargo:rerun-if-changed={}/{}.ml", OCAML_DIR, LIB_NAME);
+    println!("cargo:rerun-if-changed={}/Makefile", OCAML_DIR);
+
+    if let Some(other_dir) = env::var_os("FFI_LIB_DIR") {
+        // Link with the given library. (TODO might need to copy these in to OUT_DIR)
+        println!("cargo:rustc-link-search={}", other_dir.to_str().unwrap());
+        println!("cargo:rustc-link-lib=static={}", LIB_NAME);
+    } else {
+        // Link with generated library.
+        build_lib(out_dir, OCAML_DIR);
+        println!("cargo:rustc-link-search={}", out_dir);
+        println!("cargo:rustc-link-lib=static={}", LIB_NAME);
+    }
+}
+
+fn build_lib(out_dir: &str, ocaml_dir: &str) {
+    // Copy over the OCaml file to the out directory.
+    Command::new("make")
+        .arg(format!("BUILD_DIR={}", out_dir))
+        .current_dir(ocaml_dir)
+        .status()
+        .expect(
+            "Build of Wasm spec interpreter failed. Please ensure that you \
+        have an OCaml development environment configured as described in the \
+        wasm-spec-interpreter's README.md",
+        );
```

Comment:
Here you'll want to check the status itself (e.g. `assert!(status.success())`) because otherwise this is just successfully producing a status, but the status itself may represent an error and that's not checked.